federation/pkg/kubefed/init/init.go

```diff
@@ -121,6 +124,7 @@ func NewCmdInit(cmdOut io.Writer, config util.AdminConfig) *cobra.Command {
 	cmd.Flags().String("dns-zone-name", "", "DNS suffix for this federation. Federated Service DNS names are published with this suffix.")
 	cmd.Flags().String("image", defaultImage, "Image to use for federation API server and controller manager binaries.")
 	cmd.Flags().String("dns-provider", "google-clouddns", "Dns provider to be used for this deployment.")
+	cmd.Flags().String("dns-provider-config", "", "Config file path on local file system for configuring DNS provider.")
```

Comment:
(No action required) Do you know why the ``AddFlags()`` approach (e.g. https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-apiserver/app/options/options.go#L94) wasn't used here?  It seems cleaner to be able to use a ``FlagSet`` to define a flag and its assignment in a single line.

cc: @madhusudancs